### DISCUSSION TITLE
按照官方lora微调，使用微调后的模型出错

### DISCUSSION THREAD

**Discussion by hrq2001** [zai-org/ChatGLM3#931] [Q&A]
运行代码：python inference_hf.py output\checkpoint-3000/ --prompt "类型#裤*风格#英伦*风格#简约"
报了如下错误：

RuntimeError: CUDA error: device-side assert triggered
CUDA kernel errors might be asynchronously reported at some other API call, so the stacktrace below might be incorrect.
For debugging consider passing CUDA_LAUNCH_BLOCKING=1.
Compile with `TORCH_USE_CUDA_DSA` to enable device-side assertions.

**Comment by LiangYong1216** [comment]
解决了吗？我也是同样的问题！！！

**Comment by zyy11** [comment]
我也是同样的问题！

**Reply by zhoucan666** [reply]
一样
![Uploading 333.jpg…]()

**Comment by vw0901** [comment]
+1+1
![微信图片_20240330134435](https://github.com/THUDM/ChatGLM3/assets/55579084/2820f797-e08c-4865-8626-5b384e39c0e2)

**Comment by lowrance0118** [comment]
这个要怎么解决啊，救救孩子

**Reply by lowrance0118** [reply]
兄弟们，transformers==4.38.1，peft==0.7.1，改一下这两个库的版本，就没有这个问题了

**Reply by lukeewin** [reply]
Thank you

**Reply by yimisiyang** [reply]
adapter_config.json 这个配置文件中的参数从哪里加载进去的，参数报一个TypeError: LoraConfig.__init__() got an unexpected keyword argument 
'layer_replication'
找不到初始化了哪些参数阿

**Reply by vw0901** [reply]
我直接把他们删了，就可以运行了。不知道会不会影响效果
kwargs.pop('layer_replication')
kwargs.pop('use_dora')
kwargs.pop('use_rslora')

**Comment by zhaohai7206** [comment]
楼上的兄弟们，你们有修改什么参数的吗？我使用官方的lora微调，才几十条数据，就报CUDA out of memory，我的是2条16G的4080

**Reply by vw0901** [reply]
微调的config更新了，你调回去就可以貌似

**Reply by zhaohai7206** [reply]
可以帮忙对比看看是哪一次修改吗？我看了下lora.yaml文件的修改记录，调回去之后还是报oom
![image](https://github.com/THUDM/ChatGLM3/assets/50861760/00e1b8d3-84c0-432f-a940-c81ce4ce9a0d)

**Reply by vw0901** [reply]
![PixPin_2024-04-08_21-18-02](https://github.com/THUDM/ChatGLM3/assets/55579084/14c49655-34bf-456a-9a1a-c11c7343287c)

**Reply by zhaohai7206** [reply]
按照这个改回来了还是不行
